airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -696,10 +698,11 @@ def cleanup(self, pod: k8s.V1Pod, remote_pod: k8s.V1Pod):
                         f"Pod {pod and pod.metadata.name} returned exit code "
                         f"{self.skip_on_exit_code}. Skipping."
                     )
-            raise AirflowException(
-                f"Pod {pod and pod.metadata.name} returned a failure:\n{error_message}\n"
-                f"remote_pod: {remote_pod}"
-            )
+            raise AirflowException('\n'.join(filter(None, [
+                f"Pod {pod and pod.metadata.name} returned a failure.",
+                error_message,
+                f"remote_pod: {remote_pod}" if self.log_pod_on_failure else None
```

Comment:
I think instead of None, we could add a string here, that displaying pod specification is disabled.

Reply:
if we set it to None, we can easily filter this message out with `filter` usage. So that we'll log only messages with reasonable values.

Reply:
okay, makes sense